src/Router.tsx

```diff
@@ -20,6 +26,7 @@ export default function Router() {
         <Route path="/create">
```

Comment:
i'm not too well-read on Suspense/Lazy. any reason we shouldn't use this for v1 create as well? given i'm sure it adds a bit of loading time, we could probably omit it for projects page (which won't require loading any big ABIs anyway)

Reply:
In the case of the V1 ABIS, they're all loaded together in the `V1ContractLoader`. I did test this a while ago and recall that lazy-loading the V1 routes didn't have any effect. In an ideal world, we'd only be loading the required ABI's for each page (which would probably be a pretty huge performance boost, tbh).

I think this is where Typechain can help us eventually (https://github.com/jbx-protocol/juice-interface/issues/437). Basically it does codegen for ABI's, which I'm guessing will be tree-shakable. Definitely worth us looking into this stuff as the ABIs are by far the bulkiest thing in the bundle.

Anyways, the takeaway is that, because V2ContractLoader is only used in the V2 routes, lazy-loading the V2 routes means the V2 ABIs are _only_ loaded on V2 routes. Meaning, V1 routes aren't affected by the addition of the V2 ABI dependency. This is testable by observing the network tab: the chunk that contains the V2 ABIs is loaded when a V2 route is active:


https://user-images.githubusercontent.com/94939382/156080478-397012f8-70b6-4fa5-be8d-c36382a7f853.mp4

Reply:
got it, very cool idea.